InterfaceStubGenerator.Core/InterfaceStubGenerator.cs

```diff
@@ -114,47 +114,65 @@ internal static partial class Generated
+                                          .ToDictionary(g => g.Key, v => v.ToList());
 
             // Look through the candidate interfaces
             var interfaceSymbols = new List<INamedTypeSymbol>();
-            foreach(var iface in receiver.CandidateInterfaces)
+            foreach(var group in receiver.CandidateInterfaces.GroupBy(i => i.SyntaxTree))
             {
-                var model = compilation.GetSemanticModel(iface.SyntaxTree);
-
-                // get the symbol belonging to the interface
-                var ifaceSymbol = model.GetDeclaredSymbol(iface);
-
-                // See if we already know about it, might be a dup
-                if (ifaceSymbol is null || interfaces.ContainsKey(ifaceSymbol))
-                    continue;
-
-                // The interface has no refit methods, but its base interfaces might
-                var hasDerivedRefit = ifaceSymbol.AllInterfaces
-                                                 .SelectMany(i => i.GetMembers().OfType<IMethodSymbol>())
-                                                 .Where(m => IsRefitMethod(m, httpMethodBaseAttributeSymbol))
-                                                 .Any();
-
-                if(hasDerivedRefit)
+                var model = compilation.GetSemanticModel(group.Key);
+                foreach (var iface in group)
                 {
-                    // Add the interface to the generation list with an empty set of methods
-                    // The logic already looks for base refit methods
-                    interfaces.Add(ifaceSymbol, new List<IMethodSymbol>());
+                    // get the symbol belonging to the interface
+                    var ifaceSymbol = model.GetDeclaredSymbol(iface);
+
+                    // See if we already know about it, might be a dup
+                    if (ifaceSymbol is null || interfaces.ContainsKey(ifaceSymbol))
+                        continue;
+
+                    // The interface has no refit methods, but its base interfaces might
+                    var hasDerivedRefit = ifaceSymbol.AllInterfaces
+                                                     .SelectMany(i => i.GetMembers().OfType<IMethodSymbol>())
+                                                     .Where(m => IsRefitMethod(m, httpMethodBaseAttributeSymbol))
+                                                     .Any();
+
+                    if (hasDerivedRefit)
+                    {
+                        // Add the interface to the generation list with an empty set of methods
+                        // The logic already looks for base refit methods
+                        interfaces.Add(ifaceSymbol, new List<IMethodSymbol>() );
+                        var isAnnotated = context.Compilation.Options.NullableContextOptions == NullableContextOptions.Enable ||
+                                            model.GetNullableContext(iface.SpanStart) == NullableContext.Enabled;
```

Comment:
Is there a reason we aren't just checking the current context?

```suggestion
                        var isAnnotated = model.GetNullableContext(iface.SpanStart) == NullableContext.Enabled;
```

Reply:
I was copying the sample code you had :)